### PR TITLE
Tutorial modal

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-classifier, lib-react-components

Describe your changes:
Adds the modal UI component, with some refactors to the Modal component in lib-react-components. The styles look broken, but I anticipate will be fixed after #551. Still to do:

- [x] ~The actual PUT request to update the UPP~ (in #572)
- [x] ~Figure out why roughly 1 out 5 times of loading the page the classifier crashes because MST can't resolve the active medium~ (Fixed in #600)
- [x] Update/write tests as needed

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by srallen** [comment]
I'm getting an etag mismatch for the PUT request, so I'd like to keep this PR to just be the UI update and add the new request separately because I might have to update the panoptes.js client in the process.

**Comment by rogerhutchings** [comment]
cc #113

File: packages/lib-react-components/src/Modal/Modal.js [hunk 1]
```diff
@@ -5,11 +5,11 @@ import WithLayer from './WithLayer'
 import ModalBody from './components/ModalBody'
 import ModalHeading from './components/ModalHeading'
 
-function Modal ({ children, closeFn, colorTheme, theme, title }) {
+function Modal ({ children, className, closeFn, colorTheme, theme, title }) {
   return (
     <React.Fragment>
-      <ModalHeading closeFn={closeFn} title={title} />
-      <ModalBody colorTheme={colorTheme}>
+      <ModalHeading className={className} closeFn={closeFn} title={title} />
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/Modal/Modal.js]: should probably add propTypes for classname

File: packages/lib-react-components/src/Modal/WithLayer.js [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
-        <Layer
-          modal={modal}
-          onClickOutside={closeFn}
-          onEsc={closeFn}
-        >
-          <WrappedComponent {...props} closeFn={closeFn} />
-        </Layer>
-      </Grommet>
+      <Layer
+        className={className}
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/Modal/WithLayer.js]: and here

File: packages/lib-react-components/src/Modal/components/ModalBody/ModalBody.js [hunk 3]
```diff
@@ -4,12 +4,14 @@ import React from 'react'
...
 function ModalBody ({
   children,
+  className,
   colorTheme
 }) {
   const background = (colorTheme === 'light') ? 'white' : 'dark-5'
   return (
     <Box
       background={background}
+      className={className}
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/Modal/components/ModalBody/ModalBody.js]: and here

File: packages/lib-react-components/src/Modal/components/ModalHeading/ModalHeading.js [hunk 4]
```diff
@@ -33,11 +33,12 @@ const StyledButton = styled(Button)`
...
 `
 
-function ModalHeading ({ closeFn, title }) {
+function ModalHeading ({ className, closeFn, title }) {
   return (
     <Box
       align='center'
-      background='teal'
+      background='brand'
+      className={className}
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/Modal/components/ModalHeading/ModalHeading.js]: and here

**Comment by srallen** [comment]
My project 1233 has a tutorial. Two ways to test: signed out user and signed in user new to the project (maybe one of the zootester accounts has never visited? If not, we can update their UPPs to be like the first time)

Sometimes you might run into the reference validation error for the medium resources, but I'm aware of this and want to address it separately. If it happens, just try to refresh the page again.

**Comment by srallen** [comment]
I've updated the propTypes definitions and defaults for the className prop I added to the Modal components.

**Comment by srallen** [comment]
Also now rebased with master.

**Comment by srallen** [comment]
#572 is ready for review now.

**Comment by srallen** [comment]
#600 fixes the race condition when fetching the media and setting the modal to display.

**Comment by eatyourgreens** [comment]
When a tutorial pops up on zooniverse.org, I can change slides by pressing the cursor keys. When the tutorial pops up here, pressing the cursor keys does nothing and there's no indicator to show me where focus is.

**Comment by srallen** [comment]
@eatyourgreens the focus is broken for the tutorial itself, so it's an issue in the tabbed view as well and on master. It might be a Grommet bug because the focus is working for their `RadioButtonGroup`, so I'm looking into that. Could this be a separate issue/PR?

**Comment by srallen** [comment]
From the Grommet team, they confirm that the focus behavior is indeed different between RadioButton and RadioButtonGroup:

> RadioButtonGroup handles focus accessibly. RadioButton on it's own does not. We recommend RadioButtonGroup as the preferred means of handling radio button input. You can file an issue on RadioButton. However, I suspect it will not receive much attention from the grommet core team unless an example arises where RadioButtonGroup isn't sufficient.

I took a look at RadioButtonGroup and it's too opinionated probably because it assumes it'll be used in a form. Sounds like we'll have to either not use RadioButton from Grommet or work around it.

**Comment by eatyourgreens** [comment]
This has got pretty big. I'm happy to merge it, but keyboard accessibility must be fixed before we launch it.

**Comment by srallen** [comment]
I'll open an issue for the keyboard enhancement and see if there's a quick workaround today for it. Ideally, we'd open an issue to fix the focus in the individual RadioButton grommet component, but that's for another day.

**Pull request merged**